File: TaylorPack/doc/paper/programs/tab5/adolc/rtbp.cc
#include <adouble.h>
#include <DRIVERS/odedrivers.h>
#include <adalloc.h>

#define MU 0.01

void rtbp(int tag, double* px, double* pxp)
{
  adoublev x(6),xp(6);
  adouble r2,rpe2,rpe,rpe3i,rpm2,rpm,rpm3i,tmp;

  trace_on(tag);
  x <<= px;

  r2=x[0]*x[0]+x[1]*x[1]+x[2]*x[2];
  rpe2=r2-2*MU*x[0]+MU*MU;
  rpe=sqrt(rpe2);
  rpe3i=1/(rpe*rpe2);
  rpm2=r2+2*(1-MU)*x[0]+(1-MU)*(1-MU);
  rpm=sqrt(rpm2);
  rpm3i=1/(rpm*rpm2);
  tmp=(1-MU)*rpe3i+MU*rpm3i;

  xp[0]= x[3]+x[1];
  xp[1]= x[4]-x[0];
  xp[2]= x[5];
  xp[3]= x[4]-(1-MU)*(x[0]-MU)*rpe3i-MU*(x[0]+(1-MU))*rpm3i;
  xp[4]=-x[3]-x[1]*tmp;
  xp[5]=-x[2]*tmp;

  xp >>= pxp;
  trace_off(tag);
  return;
}

File: TaylorPack/doc/paper/programs/tab4/dop853/rtbp.f
      subroutine rtbp(n,t,x,y,rpar,ipar)
      implicit none
      integer*4 n,ipar
      real*8 t,x(6),y(6),r2,mu,d,e,f,rpe2,rpm2,rpe3i,rpm3i,rpar
      parameter (mu=0.01d0)

      r2=x(1)*x(1)+x(2)*x(2)+x(3)*x(3)
      rpe2=r2-2.d0*mu*x(1)+mu*mu
      rpe3i=1.d0/sqrt(rpe2*rpe2*rpe2)
      rpm2=r2+2.d0*(1.d0-mu)*x(1)+(1.d0-mu)*(1.d0-mu)
      rpm3i=1.d0/sqrt(rpm2*rpm2*rpm2)
      d=(1.d0-mu)*rpe3i
      e=mu*rpm3i
      f=d+e

      y(1)=x(4)+x(2)
      y(2)=x(5)-x(1)
      y(3)=x(6)
      y(4)=x(5)-(x(1)-mu)*d-(x(1)+(1.d0-mu))*e
      y(5)=-x(4)-x(2)*f
      y(6)=-x(3)*f
      return
      end
